apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarWalletSelector.tsx

```diff
@@ -0,0 +1,121 @@
+import { useCallback, useState, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import { Dropdown } from '~/components/core/Dropdown/Dropdown';
+import { DropdownItem } from '~/components/core/Dropdown/DropdownItem';
+import { DropdownSection } from '~/components/core/Dropdown/DropdownSection';
+import IconContainer from '~/components/core/IconContainer';
+import { HStack } from '~/components/core/Spacer/Stack';
+import { BaseM } from '~/components/core/Text/Text';
+import DoubleArrowsIcon from '~/icons/DoubleArrowsIcon';
+import { removeNullValues } from '~/shared/relay/removeNullValues';
+import { ChainMetadata } from '~/components/GalleryEditor/PiecesSidebar/chains';
+import { SidebarWalletSelectorFragment$key } from '~/generated/SidebarWalletSelectorFragment.graphql';
+import { useTrack } from '~/shared/contexts/AnalyticsContext';
+import { chainsMap } from '~/components/GalleryEditor/PiecesSidebar/chains';
+
+export type SidebarWallet = any;
```

Comment:
for the wallet type:
since it looks like we only need the address and chain you could define the type as 

type SidebarWallet = {
chain: string;
address: string;
}

and pass those values around instead of the whole wallet object from the graphql query.


and to account for the "All" option in the dropdown, you could either

1. define a new type that allows for either `SidebarWallet` or the string "All"

like 
type SelectedWalletOption = SidebarWallet | 'All'
and use that for the state


or 
2.
make the selectedWallet prop optional. if the "All" option is selected that means selectedWallet is not passed in